don't dump error logs in production

error dump should not be written in production, since it's not possible to retrieve them:

```
2021-03-29T23:00:01.565+09:00 | [ERROR] Exception during #32/error:report from /var/task/index.js:8133:
2021-03-29T23:00:01.565+09:00 | Error: ENOENT: no such file or directory, mkdir '/var/task/logs'
```

--- a/src/utils/dump-context.js
+++ b/src/utils/dump-context.js
@@ -13,14 +13,15 @@
 const path = require('path');
 const fs = require('fs-extra');
 const { setdefault } = require('ferrum');
+const isProduction = require('./is-production.js');
 
 /**
  * Returns {@code true} if context dumps should never be written to disk.
  * @returns {boolean} {@code true} if writing dumps is disabled.
  */
 function disableDisk() {
-  // if we run in openwhisk, we never want files to be written to disk.
-  return !!process.env.__OW_ACTIVATION_ID;
+  // if we run in production, we never want files to be written to disk.
+  return isProduction();
 }
 
 /**

--- a/src/utils/is-production.js
+++ b/src/utils/is-production.js
@@ -10,7 +10,7 @@
  * governing permissions and limitations under the License.
  */
 /**
- * Returns true when run inside an OpenWhisk action, false otherwise.
+ * Returns true when run inside an OpenWhisk action or Lambda Function, false otherwise.
  */
 function production() {
   try {

